addons/sale_timesheet/models/hr_timesheet.py

```diff
@@ -156,6 +157,21 @@ def _get_timesheets_to_merge(self):
         res = super()._get_timesheets_to_merge()
         return res.filtered(lambda l: not l.timesheet_invoice_id or l.timesheet_invoice_id.state != 'posted')
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('so_line'):
+                so_line = self.env['sale.order.line'].browse(vals['so_line'])
+                if so_line.product_id.service_policy == 'ordered_prepaid':
+                    move_id = self.env['account.move'].search([
```

Comment:
Same here. Avoid calling search in for loops.

+ ``search([('id', 'in', so_line.invoice_lines.move_id.ids)]) = so_line.invoice_lines.move_id``

You could just filter properly the results from there.